File: server/controllers/chatController.js
import Chat from '../models/Chat.js';

export const getChats = async (req, res) => {
    try {
        const chats = await Chat.find({ userId: req.userId });
        res.status(200).json(chats);
    } catch (error) {
        res.status(404).json({ message: error.message });
    }
};

export const createChat = async (req, res) => {
    const { message } = req.body;

    const newChat = new Chat({
        message,
        userId: req.userId,
        createdAt: new Date(),
    });

    try {
        await newChat.save();
        res.status(201).json(newChat);
    } catch (error) {
        res.status(409).json({ message: error.message });
    }
};

File: serverr/controllers/chat_controller.py
from flask import g, request, jsonify
from models.chat import Chat
from middleware.auth import auth_middleware
from bson import ObjectId


@auth_middleware
def get_chats():
    user_id = request.user_id
    chats = Chat.find({"user_id": ObjectId(user_id)})
    # Convert ObjectId to string
    chats_list = []
    for chat in chats:
        chat['_id'] = str(chat['_id'])
        chat['user_id'] = str(chat['user_id'])
        chats_list.append(chat)
    return jsonify(chats_list), 200

@auth_middleware
def create_chat():
    data = request.get_json()
    message = data['message']
    user_id = request.user_id

    new_chat = Chat(message=message, user_id=user_id)
    new_chat.save()
    # Convert ObjectId to string
    new_chat_dict = new_chat.__dict__
    new_chat_dict['_id'] = str(new_chat_dict['_id'])
    new_chat_dict['user_id'] = str(new_chat_dict['user_id'])
    return jsonify(new_chat_dict), 201


